backend/business-partner-agent/src/main/java/org/hyperledger/bpa/controller/PartnerController.java

```diff
@@ -93,16 +93,24 @@ public class PartnerController {
     /**
      * Get known partners
      *
-     * @param schemaId Filter Partners by schema id
+     * @param pc              {@link PaginationCommand}
+     * @param schemaId        Filter Partners by schema id
+     * @param showInvitations Filter Partners by connection state
      * @return list of partners
      */
-    @Get
-    public HttpResponse<List<PartnerAPI>> getPartners(
-            @Parameter(description = "schema id") @Nullable @QueryValue String schemaId) {
+    @Get("{?pc*}")
+    public HttpResponse<Page<PartnerAPI>> getPartners(@Valid @Nullable PaginationCommand pc,
+            @Parameter(description = "schema id") @Nullable @QueryValue String schemaId,
+            @QueryValue boolean showInvitations) {
         if (StringUtils.isNotBlank(schemaId)) {
-            return HttpResponse.ok(credLookup.getIssuersFor(schemaId));
+            List<PartnerAPI> issuersFor = credLookup.getIssuersFor(schemaId);
+            return HttpResponse
+                    .ok(Page.of(issuersFor, pc != null ? pc.toPageable() : Pageable.unpaged(), issuersFor.size()));
```

Comment:
This looks weird. Shouldn't this simply be unpaged?

Reply:
You'r right.